zerver/lib/cache.py

```diff
@@ -69,18 +76,21 @@ def get_or_create_key_prefix():
 
     return prefix
 
-KEY_PREFIX = get_or_create_key_prefix()
+KEY_PREFIX = get_or_create_key_prefix() # type: str
 
 def bounce_key_prefix_for_testing(test_name):
+    # type: (str) -> None
     global KEY_PREFIX
     KEY_PREFIX = test_name + ':' + str(os.getpid()) + ':'
 
 def get_cache_backend(cache_name):
+    # type: (str) -> get_cache
     if cache_name is None:
         return djcache
     return get_cache(cache_name)
 
 def cache_with_key(keyfunc, cache_name=None, timeout=None, with_statsd_key=None):
+    # type: ignore # TODO: https://www.python.org/dev/peps/pep-0484/#callable
```

Comment:
I think `Callable[..., Any]` is correct, but mypy doesn't like the pattern we have in models.py for creating cache keys using lambdas.  

```
@cache_with_key(lambda huddle_hash, id_list: huddle_hash_cache_key(huddle_hash), timeout=3600*24*7)                       
def get_huddle_backend(huddle_hash, id_list):                                                                             
```

I think it's OK to type:ignore this one, but we should have a clear comment about the issue, e.g.

`# type: ignore # CANNOT_INFER_LAMBDA_TYPE issue with models.py`